sriov_ib_cni_test.sh

```diff
@@ -142,35 +149,72 @@ function exit_code {
     exit $status
 }
 
-status=0
-echo "Creating pod test-pod-1"
-pod_create 'test-pod-1'
-let status=status+$?
+function test_pods {
+    status=0
 
-if [ "$status" != 0 ]; then
-    echo "Error: error in creating the first pod"
-    exit_code $status
-fi
+    echo "Creating pod test-pod-1"
+    pod_create 'test-pod-1'
+    let status=status+$?
+    
+    if [ "$status" != 0 ]; then
+        echo "Error: error in creating the first pod"
+        return $status
+    fi
+    
+    echo "Creating pod test-pod-2"
+    pod_create 'test-pod-2'
+    let status=status+$?
+    
+    if [ "$status" != 0 ]; then
+        echo "Error: error in creating the second pod"
+        return $status
+    fi
 
-echo "Creating pod test-pod-2"
-pod_create 'test-pod-2'
-let status=status+$?
+    echo ""
+    echo "Pods created successfully! VFs GUIDs after pods creation are:"
+    after_creation_guids=$(get_vfs_guids)
+    echo "$after_creation_guids"
+    
+    test_pods_connectivity 'test-pod-1' 'test-pod-2'
+     
+    let status=status+$?
+    
+    test_guids_reseted
 
-if [ "$status" != 0 ]; then
-    echo "Error: error in creating the second pod"
-    exit_code $status
-fi
+    let status=status+$?
 
-test_pods 'test-pod-1' 'test-pod-2'
+     if [ "$status" != 0 ]; then
+        echo "Error: error in testing the pods"
+        return $status
+     fi
 
-let status=status+$?
+     echo "all tests succeeded!!"
+
+     return $status
+    
+}
+
+status=0
```

Comment:
if you are already refactoring here, can you put the test flow logic under "main" and just call main

Reply:
will do